src/command_buffer.rs

```diff
@@ -169,6 +177,10 @@ impl CommandBuffer {
             let _ = world.despawn(*entity);
         }
 
+        for comp in self.commands.iter() {
```

Comment:
If this would use `self.commands.drain(..)`, then `Command` could become an `FnOnce` and consume its captures which would increase the usefulness IMHO.